### PR TITLE
Personio validation login pr

### PR DESCRIPTION
**Pull request by ArktinenSieni**
Login validation. Branch login-personio-validation-login-pr was branched from current master.

### CODE REVIEW

File: personio-client/src/Personio/Types.hs [hunk 1]
```diff
@@ -483,6 +485,31 @@ validatePersonioEmployee = withObjectDump "Personio.Employee" $ \obj -> do
...
                 i -> lift (typeMismatch "IBAN" i)
 
+        loginValidate :: WriterT [ValidationMessage] Parser ()
+        loginValidate = do
+            login <- lift (parseDynamicAttribute obj "Login name")
+            case match regexp login of
+                Nothing -> tell [LoginInvalid login]
+                Just _  -> pure ()
+          where
+              regexp :: RE' Text
```
Comment (phadej) [inline personio-client/src/Personio/Types.hs]: indentation seems to be arbitrary here

File: personio-client/src/Personio/Types.hs [hunk 2]
```diff
@@ -483,6 +485,31 @@ validatePersonioEmployee = withObjectDump "Personio.Employee" $ \obj -> do
...
+              regexp = T.pack <$> range 4 5 (psym (`elem` ['a'..'z']))
+
+              range
+                  :: Alternative f
+                  => Int  -- ^ minimum
+                  -> Int  -- ^ maximum
+                  -> f a
+                  -> f [a]
+              range mi ma f = goTo mi ma
```
Comment (phadej) [inline personio-client/src/Personio/Types.hs]: rename `goTo` to `go`

**Comment by ArktinenSieni** [comment]
Renamed the function. Does adding a line between the function and "where" make indentation look more consistent? :D

File: personio-client/src/Personio/Types.hs [hunk 3]
```diff
@@ -483,6 +485,32 @@ validatePersonioEmployee = withObjectDump "Personio.Employee" $ \obj -> do
...
                 i -> lift (typeMismatch "IBAN" i)
 
+        loginValidate :: WriterT [ValidationMessage] Parser ()
+        loginValidate = do
+            login <- lift (parseDynamicAttribute obj "Login name")
+            case match regexp login of
+                Nothing -> tell [LoginInvalid login]
+                Just _  -> pure ()
+
+          where
```
Comment (phadej) [inline personio-client/src/Personio/Types.hs]: the indentation should always be 4: 0, 4, 8. `where` is an exception, it's only half-indented:
```haskel
foo a b =
    bar a a b b  -- indentation 4, 1. stop
  where          -- indentation 2: 1/2 stop
    bar :: ...   -- indentation 4
    bar = ..
```

Same problem with `go` in `where` clause of `range`.

---

Adding empty line doesn't really help :(

**Comment by ArktinenSieni** [comment]
Took a moment to realise what was wrong with the indentation. I think I got it right this time ^^'
This time I also rebased from master.

**Comment by phadej** [comment]
Please rebase on top of current `master`

**Comment by ArktinenSieni** [comment]
Rebased @phadej

**Pull request merged**